### PR TITLE
Fixed bugs when handling errors of names of titles in the wiki  

### PR DESCRIPTION
**Pull request by RachitMalik12**
This PR fixes the following existing bugs related to error handling of wiki titles described below:

 **Bug 1**
1.  Create a new wiki page 
2.  Name it something say "Hello"
3.  Create another wiki page and name it the same as step 2 
4.  Click create and you should get the error: "Title already used"
5.  Click off by either clicking cancel or clicking outside the dialog. 
6.  Create a new page again and the error: "Title already used" pops up even though you have not entered anything 

**Bug 2**

1. Create a new wiki page. 
2.  Don't give it a title and the error "Title cannot be empty" pops up. 
3. Click off or click cancel but the error on creating a new page with a non empty title still persists. 

In short this PR fixes the bugs of the error messages not being cleared properly. 

**Testing done**
Manual testing

### CODE REVIEW

**Comment by houshuang** [comment]
@RachitMalik12 	Can you resolve merge conflicts please?

**Comment by ajainuary** [comment]
Resolved by me as the conflict was due to the changes introduced in #1723

**Comment by RachitMalik12** [comment]
@houshuang Gentle reminder to have a look at this when you have time :)

**Review by houshuang — APPROVED** [review]
